lib/node_modules/@stdlib/array/bool/lib/main.js

```diff
@@ -745,6 +746,57 @@ setReadOnly( BooleanArray.prototype, 'map', function map( fcn, thisArg ) {
 	return out;
 });
 
+/**
+* Reverses an array in-place.
+*
+* @name reverse
+* @memberof BooleanArray.prototype
+* @type {Function}
+* @throws {TypeError} `this` must be a boolean array
+* @returns {BooleanArray} reversed array
+*
+* @example
+* var arr = new BooleanArray( 3 );
+*
+* arr.set( true, 0 );
+* arr.set( false, 1 );
+* arr.set( false, 2 );
+*
+* var out = arr.reverse();
+* // returns <BooleanArray>
+*
+* var v = out.get( 0 );
+* // returns false
+*
+* v = out.get( 1 );
+* // returns false
+*
+* v = out.get( 2 );
+* // returns true
+*/
+setReadOnly( BooleanArray.prototype, 'reverse', function reverse() {
+	var buf;
+	var tmp;
+	var len;
+	var N;
+	var i;
+	var j;
+
+	if ( !isBooleanArray( this ) ) {
+		throw new TypeError( 'invalid invocation. `this` is not a boolean array.' );
+	}
+	buf = this._buffer;
+	len = this._length;
+	N = floor( len / 2 );
+	for ( i = 0; i < N; i++ ) {
+		j = len - i - 1;
+		tmp = buf[ i ];
+		buf[ i ] = buf[ j ];
+		buf[ j ] = tmp;
+	}
```

Comment:
Rather than hardcode, would it not be better to use `@stdlib/array/base/reverse` here on `this._buffer`? I.e.,

```javascript
reverse( this._buffer );
return this;
```

In short, we can fallback to the `Uint8Array#reverse` method, if it exists; otherwise, the fallback will be a manual implementation which currently implements this exact loop.